src/org/etools/j1939_84/bus/RP1210Bus.java

```diff
@@ -382,12 +353,41 @@ public void stop() throws BusException {
      * @throws BusException
      *                          if the return code is an error
      */
-    private void verify(short rtnCode) throws BusException {
+    private void checkReturnCode(short rtnCode) throws BusException {
         if (rtnCode > 127 || rtnCode < 0) {
-            rtnCode = (short) Math.abs(rtnCode);
-            byte[] buffer = new byte[256];
-            rp1210Library.RP1210_GetErrorMsg(rtnCode, buffer);
-            throw new BusException("Error (" + rtnCode + "): " + new String(buffer, StandardCharsets.UTF_8).trim());
+            String errorMessage = getErrorMessage(rtnCode);
+            throw new BusException(errorMessage);
+        }
+    }
+
+    private final AtomicBoolean reportedImposter = new AtomicBoolean(false);
```

Comment:
Should be a boolean.  It is only accessed from one thread.

Reply:
> That's a lot of code to avoid a single callback.

I agree.  I had already developed the `EventBus` and left all that in there.  I didn't want to hack in a fix.  I already didn't like that we passed the `ResultsListener` to the J1939 layer, but I didn't have a better solution at the time.